standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/client/TestPartitionStat.java

```diff
@@ -0,0 +1,304 @@
+import org.apache.hadoop.hive.metastore.utils.FileUtils;
+
+import org.apache.thrift.TException;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import com.google.common.collect.Lists;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Tests for updating partition column stats.
+ */
+@RunWith(Parameterized.class)
+@Category(MetastoreCheckinTest.class)
+public class TestPartitionStat extends MetaStoreClientTest {
+  private AbstractMetaStoreService metaStore;
+  private IMetaStoreClient client;
+  private String directSql = "false";
+
+  protected static final String DB_NAME = "test_part_stat";
+  protected static final String TABLE_NAME = "test_part_stat_table";
+  private static final String DEFAULT_COL_TYPE = "int";
+  private static final String PART_COL_NAME = "year";
+  protected static final short MAX = -1;
+  private static final Partition[] PARTITIONS = new Partition[5];
+  public static final String HIVE_ENGINE = "hive";
+
+  @BeforeClass
+  public static void startMetaStores() {
+    Map<MetastoreConf.ConfVars, String> msConf = new HashMap<MetastoreConf.ConfVars, String>();
+    // Enable trash, so it can be tested
+    Map<String, String> extraConf = new HashMap<>();
+    extraConf.put("fs.trash.checkpoint.interval", "30");  // FS_TRASH_CHECKPOINT_INTERVAL_KEY
+    extraConf.put("fs.trash.interval", "30");             // FS_TRASH_INTERVAL_KEY (hadoop-2)
+    startMetaStores(msConf, extraConf);
+  }
+
+  public TestPartitionStat(String name, AbstractMetaStoreService metaStore) {
+    this.metaStore = metaStore;
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    // Get new client, store the original value of directSql to restore it back after test.
+    directSql = metaStore.getConf().get(MetastoreConf.ConfVars.TRY_DIRECT_SQL.getVarname());
+    client = metaStore.getClient();
+
+    // Clean up the database
+    client.dropDatabase(DB_NAME, true, true, true);
+    metaStore.cleanWarehouseDirs();
+    Database db = new DatabaseBuilder().
+        setName(DB_NAME).
+        create(client, metaStore.getConf());
+
+    // Create test tables with 3 partitions
+    createTable(TABLE_NAME, getYearPartCol(), null);
+    createPartitions();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    metaStore.getConf().set(MetastoreConf.ConfVars.TRY_DIRECT_SQL.getVarname(), directSql);
+    try {
+      if (client != null) {
+        try {
+          client.close();
+        } catch (Exception e) {
+          // HIVE-19729: Shallow the exceptions based on the discussion in the Jira
+        }
+      }
+    } finally {
+      client = null;
+    }
+  }
+
+  public AbstractMetaStoreService getMetaStore() {
```

Comment:
We do not need this if we are able to create the 2 metastore instances we would like to test

Reply:
done